### PR TITLE
feat: add support for Android SDK 29 for compilation

### PR DESCRIPTION
**Pull request by KristianDD**
## PR Checklist

- [ ] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [ ] You have signed the [CLA](http://www.nativescript.org/cla).
- [ ] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [ ] Tests for the changes are included.

## What is the new behavior?

Implements issue https://github.com/NativeScript/nativescript-cli/issues/4926.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "mkdirp": "0.5.1",
     "mute-stream": "0.0.5",
     "nativescript-dev-xcode": "0.2.0",
-    "nativescript-doctor": "1.10.0",
+    "nativescript-doctor": "file:../nativescript-doctor/nativescript-doctor-1.10.0.tgz",
```
Comment (Fatme) [inline package.json]: Should be updated with the official version after the `nativescript-doctor` is released :)

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by KristianDD** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by rosen-vladimirov** [comment]
test cli-smoke cli-device cli-debug cli-run

**Comment by rosen-vladimirov** [comment]
test cli-device cli-run

**Comment by rosen-vladimirov** [comment]
test cli-device cli-run

**Comment by rosen-vladimirov** [comment]
test cli-run

**Comment by rosen-vladimirov** [comment]
test cli-run

**Pull request merged**